app/models.py

```diff
@@ -218,6 +218,45 @@ def set_fully_registered(self):
             return
         db.session.add(UserJoinedEvent.from_user(self))
 
+    def match(self, level, limit=10):
+        '''
+        Return a list of users that can answer questions this user wants to
+        learn at the specified level.
```

Comment:
Good points all.  I definitely should improve the documentation here.

The reason for the complexity of the result value is to make it easily consumable in a view.  Wrapping the complex structure of the response in some OO boilerplate ain't a bad idea.